Lombiq.Tests.UI/Extensions/MonkeyTestingUITestContextExtensions.cs

```diff
@@ -94,5 +94,54 @@ public static class MonkeyTestingUITestContextExtensions
             this UITestContext context,
             MonkeyTestingOptions options = null) =>
             new MonkeyTester(context, options).TestRecursivelyAsync();
+
+        /// <summary>
+        /// Tests the frontend (i.e. pages NOT under /admin) as monkey as an authenticated user and as an anonymous
+        /// user.
+        /// </summary>
+        /// <param name="options">The <see cref="MonkeyTestingOptions"/> instance to configure monkey testing.</param>
+        /// <param name="signInDirectlyWithUserName">
+        /// The username to sign in with directly. Defaults to <see cref="DefaultUser.UserName"/>.
+        /// </param>
+        /// <param name="startingRelativeUrl">
+        /// The relative URL to start monkey testing from. Defaults to <c>"/"</c>.
+        /// </param>
+        public static async Task TestFrontendAuthenticatedAndAnonymouslyAsMonkeyRecursivelyAsync(
+            this UITestContext context,
+            MonkeyTestingOptions options = null,
+            string signInDirectlyWithUserName = DefaultUser.UserName,
+            string startingRelativeUrl = "/")
+        {
+            await TestFrontendAnonymouslyAsMonkeyRecursivelyAsync(context, options, startingRelativeUrl);
+            await TestFrontendAuthenticatedAsMonkeyRecursivelyAsync(
+                context,
+                options,
+                signInDirectlyWithUserName,
+                startingRelativeUrl);
+        }
+
+        /// <summary>
+        /// Adds login (/Login) page to URL filters.
+        /// </summary>
+        /// <param name="options">The current <see cref="MonkeyTestingOptions"/> instance.</param>
+        public static void AddLoginPageToUrlFilters(MonkeyTestingOptions options) =>
+            options.UrlFilters.Add(new StartsWithMonkeyTestingUrlFilter("/Login"));
+
+        /// <summary>
+        /// Adds register (/Register) page to URL filters.
+        /// </summary>
+        /// <param name="options">The current <see cref="MonkeyTestingOptions"/> instance.</param>
+        public static void AddRegisterPageToUrlFilters(MonkeyTestingOptions options) =>
+            options.UrlFilters.Add(new StartsWithMonkeyTestingUrlFilter("/Register"));
+
+        /// <summary>
+        /// Adds login (/Login) and register (/Register) pages to URL filters.
+        /// </summary>
+        /// <param name="options">The current <see cref="MonkeyTestingOptions"/> instance.</param>
+        public static void AddLoginAndRegisterPagesToUrlFilters(MonkeyTestingOptions options)
+        {
+            AddLoginPageToUrlFilters(options);
+            AddRegisterPageToUrlFilters(options);
+        }
```

Comment:
Rosszul mondtam a `MonkeyTestingUITestContextExtensions`-t, inkább a `MonkeyTestingOptions`-ben kellene ezeknek metódusoknak lenniük. Érdemes fluent interface-t csinálni velük, szóval adjanak vissza `MonkeyTestingOptions`-t és akkor egymásra tudod őket chainelni híváskor.

Reply:
Ezt lehetséges hogy nem a `MonkeyTestingOptions`-be kéne írnom? Mert így akarnám, hogy chainelni lehessen:
![image](https://user-images.githubusercontent.com/80963259/153074941-d2571d8b-dbf0-4245-92c5-17899bfcc4ca.png)

`CS1106: Extension methods must be defined in a non generic static class.`

De ezt az error-t írja. Ugye ha `sealed` helyett `static` lenne az osztály, az megoldaná ezt, viszont más problémát is okozna, és akkor meg az is baj lenne hogy `static types cannot be used as return types`

Reply:
Igen, én is ezt írtam, nem extension.